python/ray/_raylet.pyx

```diff
@@ -344,42 +344,101 @@ cdef class RayletClient:
 
     def free_objects(self, object_ids, c_bool local_only, c_bool delete_creating_tasks):
         cdef c_vector[CObjectID] free_ids = ObjectIDsToVector(object_ids)
-        check_status(self.client.get().FreeObjects(free_ids, local_only, delete_creating_tasks))
+        check_status(self.client.FreeObjects(free_ids, local_only, delete_creating_tasks))
 
     def prepare_actor_checkpoint(self, ActorID actor_id):
         cdef CActorCheckpointID checkpoint_id
         cdef CActorID c_actor_id = actor_id.native()
         # PrepareActorCheckpoint will wait for raylet's reply, release
         # the GIL so other Python threads can run.
         with nogil:
-            check_status(self.client.get().PrepareActorCheckpoint(
+            check_status(self.client.PrepareActorCheckpoint(
                 c_actor_id, checkpoint_id))
         return ActorCheckpointID(checkpoint_id.Binary())
 
     def notify_actor_resumed_from_checkpoint(self, ActorID actor_id,
                                              ActorCheckpointID checkpoint_id):
-        check_status(self.client.get().NotifyActorResumedFromCheckpoint(
+        check_status(self.client.NotifyActorResumedFromCheckpoint(
             actor_id.native(), checkpoint_id.native()))
 
     def set_resource(self, basestring resource_name, double capacity, ClientID client_id):
-        self.client.get().SetResource(resource_name.encode("ascii"), capacity, CClientID.FromBinary(client_id.binary()))
+        self.client.SetResource(resource_name.encode("ascii"), capacity, CClientID.FromBinary(client_id.binary()))
 
     @property
     def language(self):
-        return Language.from_native(self.client.get().GetLanguage())
+        return Language.from_native(self.client.GetLanguage())
 
     @property
     def client_id(self):
-        return ClientID(self.client.get().GetWorkerId().Binary())
+        return ClientID(self.client.GetWorkerId().Binary())
 
     @property
     def job_id(self):
-        return JobID(self.client.get().GetJobID().Binary())
+        return JobID(self.client.GetJobID().Binary())
 
     @property
     def is_worker(self):
-        return self.client.get().IsWorker()
+        return self.client.IsWorker()
+
+
+cdef class CoreWorker:
+    cdef unique_ptr[CCoreWorker] core_worker
+
+    def __cinit__(self, is_driver, store_socket, raylet_socket, JobID job_id):
+        self.core_worker.reset(new CCoreWorker(
+            WORKER_DRIVER if is_driver else WORKER_WORKER,
+            LANGUAGE_PYTHON, store_socket.encode("ascii"),
+            raylet_socket.encode("ascii"), job_id.native()))
+
+    # TODO: reconstructions
+    def get_objects(self, object_ids, deserialization_context=None):
+        cdef c_vector[CObjectID] get_ids = ObjectIDsToVector(object_ids)
+        cdef c_vector[shared_ptr[CRayObject]] results
+
+        timeout = RayConfig.instance().get_timeout_milliseconds()
+        check_status(self.core_worker.get().Objects().Get(get_ids, timeout, &results))
```

Comment:
Probably we still need to split the object_id list into chunks as it was before. Otherwise the raylet might be blocked on a large Get, you should try it out.